### ISSUE TITLE
Feature request: Day-by-day visualization for all activities

### ISSUE THREAD

**Issue by philip-n** [babybuddy/babybuddy#218]
To me, one of the key benefits of storing data / information in a structured way is the possibility to analyze them later, and I found that visualizations can be of great help.

I really like the sleep rhythm visualization and would love to have a similar (identical) general visualization that displays various (selectable) activities.

Examples for what could be displayed and what the user might learn:
* Sleep times and feeding times - "Baby slept longer after being fed longer"
* Feeding times and diaper changes - "changes mostly occurred 2h after feeding"
* Display a single activity and see its distribution over the day

**Comment by cephos** [comment]
I had a similar requirement and built a timeline-graph on top of the api. Do you mean something like this?

![BabybuddyTimeline](https://user-images.githubusercontent.com/1743497/119772844-188faf00-bec0-11eb-9a9e-31aa200523d5.jpg)

If yes, i could clean up the code a bit and publish it here (needs some reverse proxy-magic because of CORS limitations, but I can describe that as well). Perhaps someone with a bit more django experience than me wants to integrate that into babybuddy so the reverse-proxy can be left out.

**Comment by cdubz** [comment]
Interesting @cephos -- depending on the complexity of this it make sense to keep it as a separate package of sorts that uses the API. If so, we could also publish it to the Baby Buddy org on GH. Happy to see the code when you have some time to get it posted.

In general I would _love_ to have more advanced graphing tools but its just not an area where I am able to spend a lot of time right now (and it seems like it would take some time). The sleep pattern graph is really nice, but it's also kind of insanely complicated. As I think about it, it would be nice to offer a separate package for data visualization that works off the API without being part of the main app. I think that would make it more maintainable and useful to folks who really want that depth of detail.

**Comment by philip-n** [comment]
That's a nice visualization @cephos ! It is very similar to what I have in mind, indeed. Would be great if you could publish your code! Is it custom-built or built on top of some framework/library?
Also, I'm just a little bit jealous of the depicted sleeping patterns at night ;)

Maintaining this as a separate package sounds very reasonable, as @cdubz suggests. As added benefit, it would allow development of the viz package on top of a "default" installation of the BabyBuddy core, without the requirement to install the BabyBuddy development environment.

Only thing is that the installation would need to be "easy" - if it is too complicated, one could just install Grafana instead and point it to the BabyBuddy API.

**Comment by JeanFred** [comment]
Yeah, I would have loved something like the visualisation done at https://web.archive.org/web/20170724003103/https://www.pagerduty.com/blog/babyduty/ − https://www.pagerduty.com/wp-content/uploads/2017/06/babyduty_timeline.png :) (The sleep patterns graph comes close, but not quite).
